File: backend/src/main/java/com/ufc/backend/backend/model/feedback/Feedback.java
package com.ufc.backend.backend.model.feedback;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.ufc.backend.backend.model.Classes;
import com.ufc.backend.backend.model.Student;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Feedback {

    @Id
    private String id;
    private String commentary;
    private Rating rating;
    @DBRef
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Classes classes;
    private Student user;
    @JsonIgnore
    private Student actualUser;
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private boolean isAnonymous;

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        if (getClass() != obj.getClass()) {
            return false;
        }
        return this.getId().equals(((Feedback) obj).getId());
    }
}

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
class Rating {

    private double howDifficultItWasToLearn;
    private double isThereALotOfWorksToDo;
    private int howManyExams;
}

File: backend/src/main/java/com/ufc/backend/backend/model/feedback/Feedback.kt
package com.ufc.backend.backend.model.feedback

import com.ufc.backend.backend.commons.model.Identifiable
import com.ufc.backend.backend.model.Subject
import com.ufc.backend.backend.model.Student
import org.hibernate.annotations.GenericGenerator
import javax.persistence.*

@Entity
class Feedback
(
    @Id
    @GeneratedValue(generator = "uuid2")
    @GenericGenerator(name = "uuid2", strategy = "org.hibernate.id.UUIDGenerator")
    @Column
    override val id: String,
    @Column
    var commentary: String,
    @Column
    var rating: Double,
    @OneToOne
    @JoinColumn(name = "subject_id", referencedColumnName = "id")
    var subject: Subject,
    @OneToOne
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    var user: Student,
    var isAnonymous: Boolean = false,
) : Identifiable<String>